nemo_rl/models/generation/vllm/vllm_worker.py

```
@@ -333,6 +334,21 @@ def _patch_vllm_sampler():
 
+        # We should use vLLM DP if ep_size > tp_size since EP_SIZE = DP_SIZE * TP_SIZE in vLLM.
+        # See details in https://github.com/vllm-project/vllm/blob/main/examples/offline_inference/data_parallel.py
+        if self.expert_parallel_size > self.tensor_parallel_size:
+            # set vLLM DP rank
+            world_size = int(os.environ["VLLM_DP_SIZE"]) * model_parallel_size
+            rank = int(os.environ["RANK"]) % world_size
+            os.environ["VLLM_DP_RANK"] = str(rank // model_parallel_size)
+            os.environ["VLLM_DP_RANK_LOCAL"] = str((rank % 8) // model_parallel_size)
+            # set vLLM DP address and port
+            leader_rank = int(os.environ["RANK"]) // world_size * world_size
+            addr_list = eval(os.environ["AVAILABLE_ADDR_LIST"])
```

Ruff (0.12.2)

Use of possibly insecure function; consider using `ast.literal_eval`

(S307)